packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts

```diff
@@ -49,34 +56,75 @@ export default class BulkExtractor {
         if (state) {
           states.push(state);
         } else {
-          const runState = await this.getSingleState(runId, await this._accessToken());
+          const runState = await this.getState(runId, await this._accessToken());
           states.push(runState);
           stateByRunId.set(runId, runState);
         }
       }
       const finalState = BulkExtractor.getFinalState(states);
       stateByReportId.set(reportId, finalState);
     }
-    return stateByReportId;
-  }
 
-  private async fetchStates(): Promise<void> {
-    this._reportStates = await this.getStates(this._reportIds);
+    for (const iModelRun of this._iModelRun) {
+      let state = stateByRunId.get(iModelRun[1]);
+
+      if (!state) {
+        state = await this.getState(iModelRun[1], await this._accessToken());
+        stateByRunId.set(iModelRun[1], state);
+      }
+
+      const finalState = BulkExtractor.getFinalState([state]);
+      stateByIModelId.set(iModelRun[0], finalState);
+    }
+
+    this._reportStates = stateByReportId;
+    this._iModelStates = stateByIModelId;
   }
 
-  public getState(reportId: string): ExtractionStates {
+  public async getReportState(reportId: string): Promise<ExtractionStates> {
     if ((new Date().getTime() - this._timeFetched.getTime()) > STATUS_CHECK_INTERVAL) {
       this._timeFetched = new Date();
-      this.fetchStates().catch((e) =>
+      await this.fetchStates().catch((e) =>
         /* eslint-disable no-console */
         console.error(e)
       );
     }
     return this._reportStates.get(reportId) ?? ExtractionStates.None;
   }
 
-  public clearJob(reportId: string): void {
+  public async getIModelState(iModelId: string, iModelName: string, odataFeedUrl: string): Promise<ExtractionStates> {
+    let state = this._iModelStates.get(iModelId) ?? ExtractionStates.None;
+
+    if (!(state === ExtractionStates.Succeeded || state === ExtractionStates.Failed)) {
+      if ((new Date().getTime() - this._timeFetched.getTime()) > STATUS_CHECK_INTERVAL) {
+        this._timeFetched = new Date();
+        await this.fetchStates().catch((e) =>
+          /* eslint-disable no-console */
+          console.error(e)
+        );
+        state = this._iModelStates.get(iModelId) ?? ExtractionStates.None;
+      }
+    }
+    if (state === ExtractionStates.Succeeded && !this._iModelToast.get(iModelId)) {
+      this._successfulExtractionToast(iModelName, odataFeedUrl);
+      this._iModelToast.set(iModelId, true);
+    } else if (state === ExtractionStates.Failed && !this._iModelToast.get(iModelId)) {
+      this._failedExtractionToast(iModelName);
+      this._iModelToast.set(iModelId, true);
+    }
+    return state;
+  }
+
+  public clearReportJob(reportId: string): void {
```

Comment:
Looks like this method clears a lot more than just the reportJob. Why is this needed?

Reply:
`BulkExtractor` refactored so this method is no longer needed and has been removed.